core/src/main/scala/org/apache/spark/api/r/RRunner.scala

```diff
@@ -125,7 +125,14 @@ private[spark] class RRunner[IN, OUT](
               eos = true
               null.asInstanceOf[OUT]
           }
-        } catch handleException
+        } catch {
+          case e: EOFException =>
```

Comment:
This EOFException is from L98.

Reply:
How can we be confident that this is the only cause for `EOFException`. If for any other reason we throw this exception, your handling here would silently ignore it and return a null value, which could become very confusing to users.